src/sa/sa.c

```diff
@@ -17,6 +20,30 @@
 #define DEBUG_LEVEL 5
 #include <re_dbg.h>
 
+static int resolve_address(const char *addr, struct sa *sa)
```

Comment:
should we move this to net.c and make it generic ?

Reply:
Good idea!

Reply:
`getaddrinfo()` also does a DNS lookup if the format of addr is not an IP address. The timeout can be set in resolve.conf and is by default 5 seconds. The DNS lookup is very fast if the DNS server is reachable, even if the queried address is not reachable. Anyway, we don't want a synchronous DNS lookup.

Reply:
Should we add a check in `net_resolve_address()`:
```
@@ -236,6 +236,10 @@ int net_resolve_address(const char *addr, struct sa *sa)
        struct addrinfo *res, *res0;
        int err;
 
+       err = net_inet_pton(addr, sa);
+       if (err)
+               return err;
+
        err = getaddrinfo(addr, NULL, NULL, &res0);
        if (err) {
                DEBUG_WARNING("getaddrinfo(): (%s)\n", gai_strerror(err));
```

For this we would have to move `net_inet_pton()` from internal sa.h to re_sa.h.

Reply:
Or we rename the function to `net_resolve_ipv6ll()` and move the simple check which is now in sa.c to `net_resolve_ipv6ll()`:
```
if (!strncmp(addr, "fe80:", 5))
```